File: Python/0048. Domino piling.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            m,n = map(int,input().strip().split())
            if m%2 == 0 or n%2 == 0:
                print(m*n//2)
            else:
                print(m*(n//2)+m//2)
            
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0048. Domino piling.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int m,n;
    while(cin>>m>>n){
        if(m%2==0 || n%2==0){
            cout<<m*n/2<<endl;
        }
        else{
            cout<<m*(n/2)+m/2<<endl;
        }
    }
    return 0;
}
